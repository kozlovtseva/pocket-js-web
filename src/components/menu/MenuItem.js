import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import { logoutUser } from '../../store/actions';
import { connect } from 'react-redux';

import styles from './MenuItem.module.css';

export class MenuItem extends Component {

	onClick = (text) => {
		if(text === 'Выйти'){
			this.props.dispatch(logoutUser());
		}else if(text === 'Черный список'){
			this.props.usersListToggle();
		}
	}

	render() {
		let item = (this.props.href === null) ? (
			<div className={styles.Item} onClick={() => {this.onClick(this.props.text)}}>
				<i className={styles.Icon + this.props.icon} />
				<p className={styles.Text}>{this.props.text}</p>
			</div>
		) :
			<Link to={this.props.href} className={styles.Item} onClick={() => {this.onClick(this.props.text)}}>
				<i className={styles.Icon + this.props.icon} />
				<p className={styles.Text}>{this.props.text}</p>
			</Link>
		return (
			<>
				{item}
			</>
		)
	}
}

export default connect()(MenuItem);

