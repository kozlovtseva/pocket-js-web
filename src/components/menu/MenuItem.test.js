import React from 'react';
import '../../../setupTests';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import { MenuItem } from './MenuItem';


const props = {
  icon: ' fas fa-user-times',
  text: 'Выйти',
  href: '/auth',
};
describe('render MenuItem', () => {
  it('render correctly MenuItem component', () => {
    const MenuItemComponent = renderer.create(<Router><MenuItem {...props} /></Router>).toJSON();
    expect(MenuItemComponent).toMatchSnapshot();
  });
  it('render with props correctrly (link)', () => {
    const MenuItemComponent = mount(<Router><MenuItem {...props} /></Router>);
    expect(MenuItemComponent.find('a').length).toEqual(1);
  });
  it('render with props correctrly (not link)', () => {
    const propsForDiv = {
      icon: ' fa-user-slash',
      text: 'Черный список',
      href: null,
    };
    const MenuItemComponent = mount(<Router><MenuItem {...propsForDiv} /></Router>);
    expect(MenuItemComponent.find('div').length).toEqual(1);
  });
});
