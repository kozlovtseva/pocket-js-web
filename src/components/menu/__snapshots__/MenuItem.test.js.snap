// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render MenuItem render correctly MenuItem component 1`] = `
<a
  className="Item"
  href="/auth"
  onClick={[Function]}
>
  <i
    className="Icon fas fa-user-times"
  />
  <p
    className="Text"
  >
    Выйти
  </p>
</a>
`;
